src/create-element.js

```diff
@@ -18,6 +18,13 @@ export function createElement(type, props, children) {
 		key,
 		ref,
 		i;
+
+	// @ts-expect-error type can't be null, however TS is confused
+	if (type != null && typeof type === 'object' && 'Provider' in type) {
```

Comment:
I wonder whether we should move this to compat instead, this looks like some overhead in a very hot path

Reply:
Given the pretty minimal utility, I think I'd vote for/prefer moving to compat myself.